src/main/webapp/app/exercises/modeling/manage/modeling-exercise-update.component.ts

```diff
@@ -211,27 +225,36 @@ export class ModelingExerciseUpdateComponent implements OnInit {
         );
     }
 
-    /**
-     * Revert to the previous state, equivalent with pressing the back button on your browser
-     * Returns to the detail page if there is no previous state
-     * Returns to the overview page if there is no previous state and we created a new exercise
-     * Returns to the exercise groups page if we are in exam mode
-     */
     previousState() {
         navigateBackFromExerciseUpdate(this.router, this.modelingExercise);
     }
 
     private subscribeToSaveResponse(result: Observable<HttpResponse<ModelingExercise>>): void {
         result.subscribe(
-            () => this.onSaveSuccess(),
+            (exercise: HttpResponse<ModelingExercise>) => this.onSaveSuccess(exercise.body!.id!),
             () => this.onSaveError(),
         );
     }
 
-    private onSaveSuccess(): void {
+    private onSaveSuccess(exerciseId: number): void {
         this.eventManager.broadcast({ name: 'modelingExerciseListModification', content: 'OK' });
         this.isSaving = false;
-        this.previousState();
+
+        switch (this.editType) {
+            case EditType.CREATE:
+                // Assigning id received from the backend.
+                // Required for navigation to the example submission dashboard in case of the exercise CREATE.
```

Comment:
nit: I think "in case of the exercise CREATE" is not really necessary, as this is indicated by `case EditType.CREATE` and if you still want to keep it, I would write "in case the exercise is newly created"

Reply:
done